src/DotLiquid.Tests/CustomIndexableTests.cs

```diff
@@ -21,8 +21,8 @@ internal VirtualList(params object[] items)
             {
                 get
                 {
-                    int? index = key as int?;
-                    if (!index.HasValue || index.Value < 0 || index.Value >= items.Length) {
+                    long? index = key as long?;
```

Comment:
While long are implemented array / index accessors in .NET, it works in conjunction with more common int index keys. So we need at minimum to keep supporting int index and optional support long indexes as well.